packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -1849,18 +1874,33 @@ class JavaGenerator extends Generator {
     const props = Object.values(propsByName);
     this.emitInterfaceBuilder(ifc, INTERFACE_PROXY_CLASS_NAME, props);
 
+    // If the data type has exactly 1 base datatype, we'll extend the base's proxy
+    const baseInterface =
+      ifc.interfaces?.length === 1 ? ifc.interfaces[0] : undefined;
+    const ownProps = props.filter(
+      (prop) => baseInterface == null || !prop.inherited,
```

Comment:
This is stretching the definition of `ownProps` somewhat, in my opinion.

I mean, I guess it's "own" for *someone* (the generated class or the *generatee* interface), but this is not the someone I would have expected. I would have expected "own" to mean "as declared", not "as generated".

Maybe call it `hereProps`/`thereProps` instead? (For funzies)

Or `declaredProps`/`inheritedProps` ?

Reply:
Yeah I guess it's "contextually own props" sort of... Can definitely be misleading when looking only at usage sites...